.github/workflows/test.yml

```diff
@@ -1,39 +1,118 @@
-name: Test
-"on":
+name: CI
+on:
+  workflow_dispatch:
   push:
     branches:
       - main
-      - dependabot/npm_and_yarn/**
   pull_request:
     types:
       - opened
       - synchronize
+
+permissions:
+  contents: read
+
 jobs:
-  test_matrix:
+  dependency-review:
+    name: Dependency Review
+    if: github.event_name == 'pull_request'
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node_version:
-          - 10
-          - 12
-          - 14
     steps:
-      - uses: actions/checkout@v2
-      - name: Use Node.js ${{ matrix.node_version }}
-        uses: actions/setup-node@v2
+      - name: Check out repository
+        uses: actions/checkout@v4
         with:
-          node-version: ${{ matrix.node_version }}
-          cache: npm
-      - run: npm ci
-      - run: npx tap 'test/*.js'
-  test:
+          persist-credentials: false
+
+      - name: Dependency review
+        uses: actions/dependency-review-action@v4
+
+  license-check:
+    name: Check Licenses
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+      - name: Install dependencies
+        run: npm ci
+      - name: Check Licenses
+        run: npx license-checker --production --summary --onlyAllow="0BSD;Apache-2.0;Apache 2.0;Python-2.0;BSD-2-Clause;BSD-3-Clause;ISC;MIT"
+
+  codeql:
+    name: CodeQL
     runs-on: ubuntu-latest
-    needs: test_matrix
+    timeout-minutes: 360
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: javascript-typescript
+          queries: security-and-quality
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v3
+        with:
+          category: "/language:javascript-typescript"
+
+  lint:
```

Comment:
There needs to be a wrapper job so we don't require specific node versions I'm branch protection rules.

Reply:
what do you mean?

Reply:
Like this
https://github.com/probot/probot/blob/880f2b93160e88bb9f5cdbd336c2b68702aa51f1/.github/workflows/test.yml#L118-L122

Reply:
I added the wrapper, but i dont see the point